doc/cabal-project.rst

```diff
@@ -1333,156 +1336,140 @@ Haddock options
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-benchmarks: boolean
+               --haddock-benchmarks
     :synopsis: Generate documentation for benchmarks.
 
     :default: False
 
     Run haddock on all benchmarks.
 
-    The command line variant of this flag is ``--benchmarks`` (for the
-    ``haddock`` subcommand).
-
-.. cfg-field:: haddock-all: boolean
-    :synopsis: Generate documentation for everything
-
-    :default: False
-
-    Run haddock on all components.
-
-    The command line variant of this flag is ``--all`` (for the
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-internal: boolean
+               --haddock-internal
     :synopsis: Generate documentation for internal modules
 
     :default: False
 
     Build haddock documentation which includes unexposed modules and
     symbols.
 
-    The command line variant of this flag is ``--internal`` (for the
-    ``haddock`` subcommand).
+.. cfg-field:: haddock-all: boolean
+               --haddock-all
+    :synopsis: Generate documentation for everything
+
+    :default: False
+
+    Run haddock on all components.
 
 .. cfg-field:: haddock-css: path
-    :synopsis: Location of Haddoc CSS file.
+               --haddock-css=PATH
+    :synopsis: Location of Haddock CSS file.
 
     The CSS file that should be used to style the generated
-    documentation (overriding haddock's default.)
-
-    The command line variant of this flag is ``--css`` (for the
-    ``haddock`` subcommand).
+    documentation (overriding haddock's default).
 
 .. cfg-field:: haddock-hyperlink-source: boolean
+               --haddock-hyperlink-source
     :synopsis: Generate hyperlinked source code for documentation
 
     :default: False
 
     Generated hyperlinked source code using `HsColour`_, and have
     Haddock documentation link to it.
-
-    The command line variant of this flag is ``--hyperlink-source`` (for
-    the ``haddock`` subcommand).
+    This is equivalent to running ``haddock`` with the ``--hyperlinked-source`` flag.
 
 .. cfg-field:: haddock-hscolour-css: path
+               --haddock-hscolour-css=PATH
     :synopsis: Location of CSS file for HsColour
 
     The CSS file that should be used to style the generated hyperlinked
     source code (from `HsColour`_).
 
-    The command line variant of this flag is ``--hscolour-css`` (for the
-    ``haddock`` subcommand).
-
 .. cfg-field:: haddock-contents-location: URL
+               --haddock-contents-location=URL
     :synopsis: URL for contents page.
 
     A baked-in URL to be used as the location for the contents page.
 
-    The command line variant of this flag is ``--contents-location``
-    (for the ``haddock`` subcommand).
-
 .. cfg-field:: haddock-keep-temp-files: boolean
     :synopsis: Keep temporary Haddock files.
 
     Keep temporary files.
 
-    The command line variant of this flag is ``--keep-temp-files`` (for
-    the ``haddock`` subcommand).
+    There is no command line variant of this flag.
```

Comment:
Was it removed recently? I seem to recall people discussing it recently on #haskell. Anyway, if you’re sure that it’s gone, than it’s good.

Reply:
It's gone, apparently, and this is very easy to check:
```
$ cabal haddock --keep-temp-files
Error: cabal: unrecognized 'haddock' option `--keep-temp-files'
$ cabal haddock --haddock-keep-temp-files
Error: cabal: unrecognized 'haddock' option `--haddock-keep-temp-files'
```

Reply:
That said, it confuses me that it exists as a `cabal.project` field, _only_, and it does not seem an option that is simply passed to `haddock`, because `haddock` does not seem to have an equivalent one.

Reply:
_Keeping temp files_ seems to be a Cabal thing rather than a haddock thing:
https://github.com/haskell/cabal/blob/63841fb3380b902d12118fc272b5421a923cd647/Cabal/src/Distribution/Simple/Utils.hs#L1438-L1439

That satisfies my curiosity.